### ISSUE TITLE
missing diagnostics for some F#, C# code

### ISSUE THREAD

**Issue by dsyme** [dotnet/interactive#691]
Put this into a notebook and execute:

    open Microsof  t.DotN et

Expected: an error 
Actual: no error is shown 

Tested with https://mybinder.org/v2/gh/dotnet/interactive/master?urlpath=lab

**Comment by dsyme** [comment]
There are other examples where errors aren't shown, e.g. evaluate

```
1+
```

or

```
1+!
```

**Comment by dsyme** [comment]
Also noticed missing warnings diagnostics from C# code, e.g. System.Obsolete messages

**Comment by dsyme** [comment]
@brettfo The problem here is thar FsiScript.Eval is succeeding with `Ok` even when there are error diagnostics from parsing and type checking:
```
            let result, diagnostics =
                try
                    script.Value.Eval(codeSubmission.Code, tokenSource.Token)
                with
                | ex -> Error(ex), [||]
```

I'll fix this up in .NET Interacive to double check for an Error severity diagnostic.  However I think we should also fix this in the scripting DLL shim in dotnet/fsharp.

**Comment by dsyme** [comment]
A repro for missing errors in F#

```
1+!
```

A repro for missing warnings in both F# and C# (using an Obsolete API)

```
System.AppDomain.GetCurrentThreadId();
```

**Comment by dsyme** [comment]
BTW the difference between these is very subtle in C# :-)
```
System.AppDomain.GetCurrentThreadId()
```
and
```
System.AppDomain.GetCurrentThreadId();
```
The first reveals an integer answer, the second the result is implicitly thrown away because of C#'s crazy "ignore this expression" rule.

**Comment by dsyme** [comment]
Fixed by #713
